aten/src/ATen/native/Sorting.cpp

```diff
@@ -271,8 +267,124 @@ std::tuple<Tensor&, Tensor&> kthvalue_out_impl_cpu(
+    } else {
+      out_shape.erase(out_shape.begin() + dim);
+    }
+  }
+
+  resize_output(values, out_shape);
+  resize_output(indices, out_shape);
+
+  Tensor vals = keepdim ? values : values.unsqueeze(dim);
+  Tensor inds = keepdim ? indices : indices.unsqueeze(dim);
+
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "median_out", [&] {
+    dim_apply({self, vals, inds}, dim, [ignore_nan](int64_t it, TensorList tl) {
+      Tensor in = tl[0].contiguous();
+      scalar_t* ip = in.data_ptr<scalar_t>();
+      int64_t size = in.numel();
+
+      if (!ignore_nan) {
+        scalar_t* nanp = std::find_if(ip, ip + size, _isnan<scalar_t>);
+        if (nanp != ip + size) {
+          *tl[1].data_ptr<scalar_t>() = *nanp;
+          *tl[2].data_ptr<int64_t>() = nanp - ip;
+          return;
+        }
+      }
+
+      std::vector<int64_t> idx(size);
+      auto first = idx.begin();
+      auto last = idx.end();
+      std::iota(first, last, 0);
+
+      auto nth = first;
+      if (!ignore_nan) {
+        nth += (size - 1) / 2;
+        std::nth_element(first, nth, last, [&ip](int64_t i, int64_t j) {
+          return ip[i] < ip[j] || (ip[i] == ip[j] && i < j);
+        });
+      } else {
+        int64_t num_nan = std::count_if(ip, ip + size, _isnan<scalar_t>);
+        nth += (size - num_nan - 1) / 2;
+        std::nth_element(first, nth, last, [&ip](int64_t i, int64_t j) {
+          return ip[i] < ip[j] || (ip[i] == ip[j] && i < j) ||
+              (_isnan(ip[j]) && !_isnan(ip[i]));
+        });
+      }
+
+      *tl[1].data_ptr<scalar_t>() = ip[*nth];
+      *tl[2].data_ptr<int64_t>() = *nth;
+    });
+  });
+
+  guard.reset();
+  namedinference::propagate_names_for_reduction(values, self, dim, keepdim);
+  namedinference::propagate_names_for_reduction(indices, self, dim, keepdim);
+
+  return std::forward_as_tuple(values, indices);
+}
+
+Tensor median_impl(const Tensor& self, bool ignore_nan) {
+  NoNamesGuard guard;
+
+  int64_t size = self.numel();
+  TORCH_CHECK(size > 0, "median() input tensor cannot be empty");
+
+  Tensor in = self.clone();
+  Tensor out = at::empty({}, self.options());
+
+  AT_DISPATCH_ALL_TYPES(in.scalar_type(), "median_cpu", [&] {
+    scalar_t* op = out.data_ptr<scalar_t>();
+    scalar_t* first = in.data_ptr<scalar_t>();
+    scalar_t* last = first + size;
+
+    if (!ignore_nan && std::any_of(first, last, _isnan<scalar_t>)) {
+      *op = std::numeric_limits<scalar_t>::quiet_NaN();
+      return;
+    }
+
+    scalar_t* median = first;
+    if (!ignore_nan) {
+      median += (size - 1) / 2;
+      std::nth_element(first, median, last);
+    } else {
+      int64_t num_nan = std::count_if(first, last, _isnan<scalar_t>);
+      median += (size - num_nan - 1) / 2;
+      std::nth_element(first, median, last, [](scalar_t a, scalar_t b) {
+        return a < b || (_isnan(b) && !_isnan(a));
+      });
+    }
+
+    *op = *median;
+  });
+
+  return out;
+}
+
 } // namespace
 
+// MARK : quantile
```

Comment:
What does "MARK :" mean?

Reply:
This gives you some bookmarking in Xcode, but I guess it's confusing. I'll change it to a regular comment.